onadata/apps/logger/management/commands/update_attachment_storage_bytes.py

```diff
@@ -0,0 +1,147 @@
+
+                # Set the flag to true if it was never set.
+                if not user_profile.metadata.get('submissions_suspended'):
+                    # We are using the flag `submissions_suspended` to prevent
+                    # new submissions from coming in while the
+                    # `attachment_storage_bytes` is being calculated.
+                    user_profile.metadata['submissions_suspended'] = True
+                    user_profile.save(update_fields=['metadata'])
+
+                # if `last_xform` is not none, it means that the `user_id` is
+                # different from the previous one in the loop so the user
+                # profile must be updated
+                if last_xform:
+                    self.update_user_profile(
+                        last_xform, total_per_user[last_xform['user_id']]
+                    )
+
+            # write out xform progress
+            if self.verbosity >= 1:
+                self.stdout.write(
+                    f"Calculating attachments for xform_id #{xform['pk']}"
+                    f" (user {xform['user__username']})"
+                )
+            # aggregate total media file size for all media per xform
+            form_attachments = Attachment.objects.filter(
+                instance__xform_id=xform['pk']
+            ).aggregate(total=Sum('media_file_size'))
+
+            if form_attachments['total']:
+                with transaction.atomic():
+                    if self.verbosity >= 1:
+                        self.stdout.write(
+                            f'\tUpdating xform attachment storage to '
+                            f"{form_attachments['total']} bytes"
+                        )
+
+                    XForm.objects.filter(
+                        pk=xform['pk']
+                    ).update(
+                        attachment_storage_bytes=form_attachments['total']
+                    )
+
+                total_per_user[xform['user_id']] += form_attachments['total']
+            elif self.verbosity >= 1:
+                self.stdout.write('\tNo attachments found')
+
+            last_xform = xform
+
+        # need to call `update_user_profile()` one more time outside the loop
+        # because the last user profile will not be up-to-date otherwise
+        if last_xform:
+            self.update_user_profile(
+                last_xform, total_per_user[last_xform['user_id']]
+            )
+
+        if self.verbosity >= 1:
+            self.stdout.write('Done!')
+
+    def update_user_profile(self, xform: dict, total: int):
+        user_id = xform['user_id']
+        username = xform['user__username']
+
+        if self.verbosity >= 1:
+            self.stdout.write(
+                f'Updating attachment storage total ({total} bytes) on '
+                f'{username}’s profile'
+            )
+
+        # Update user's profile (and lock the related row)
+        with transaction.atomic():
```

Comment:
I don't know why I wrote that I understood this [before](https://github.com/kobotoolbox/kobocat/pull/828/files/f740202660d6a2cc14b647c5a357cb39f1f736a9#diff-d3f422f136f8d6948d5f29625f3772f955dc4d8f96a492885c0b494054aed21d) :see_no_evil:
`update()` is already atomic; why do we need `transaction.atomic()` too?

Reply:
gone!